src/beneuro_data/cli.py

```diff
@@ -678,6 +742,119 @@ def upload_session(
+            "-v/-V",
+            help="Download video data (-v) or not (-V)."
+        ),
+    ] = False,
+    include_extra_files: Annotated[
+        bool,
+        typer.Option(
+            "--include-extra-files/--ignore-extra-files",
+            help="Upload extra files that are created by the experimenter or other software.",
+        ),
+    ] = True,
+    rename_videos_first: Annotated[
+        bool,
+        typer.Option(
+            "--rename/--no-rename",
+            help="Rename videos before validating and uploading.",
+        ),
+    ] = True,
+    rename_extra_files_first: Annotated[
+        bool,
+        typer.Option(
+            "--rename-extra/--no-rename-extra",
+            help="Rename extra files (e.g. comment.txt) before validating and uploading.",
+        ),
+    ] = True,
+    processing_level: Annotated[
+        str, typer.Argument(help="Processing level of the session. raw or processed.")
+    ] = "raw",
+):
+    """
+    Upload (raw) experimental data to the remote server.
+
+    Example usage to upload everything of a given session:
+        `bnd up M017_2024_03_12_18_45 -ev`
+    Example to upload the videos and ephys of the last session of a subject:
+        `bnd up M017 -evB`
+    """
+    animal = session_name[:4]
+
+    if processing_level != "raw":
+        raise NotImplementedError("Sorry, only raw data is supported for now.")
+
+    if all([not include_behavior, not include_ephys, not include_videos]):
+        raise ValueError("At least one data type must be included.")
+
+    # if videos are included, rename them first if not specified otherwise
+    if rename_videos_first is None:
+        rename_videos_first = include_videos
+
+    if rename_videos_first and not include_videos:
+        raise ValueError(
+            "Do not rename videos if you're not uploading them. (Meaning --ignore-videos and --rename-videos are not allowed together.)"
+        )
+
+    config = _load_config()
+
+    if len(session_name) > 4:  # session name is given
+        up_done = upload_raw_session(
+            config.LOCAL_PATH / processing_level / animal / session_name,
+            animal,
+            config.LOCAL_PATH,
+            config.REMOTE_PATH,
+            include_behavior,
+            include_ephys,
+            include_videos,
+            include_extra_files,
+            config.WHITELISTED_FILES_IN_ROOT,
+            config.EXTENSIONS_TO_RENAME_AND_UPLOAD,
+            rename_videos_first,
+            rename_extra_files_first,
+        )
+    else:  # only animal name is given
+        up_done = upload_last(
+            animal,
+            include_behavior,
+            include_ephys,
+            include_videos,
+            include_extra_files,
+            rename_videos_first,
+            rename_extra_files_first,
+            processing_level,
+        )
+    if up_done is True:
+        print(f"Session {session_name} uploaded successfully.")
```

Comment:
Without my suggestions, `session_name` could be an animal name here if that was passed.
With my suggestions it would be `None`.
So if an animal name was passed, either print something like `f"Last session of {animal} uploaded successfully."`, or figure out what that last session was. It's done like this in `upload_last`:
```
subject_path = config.LOCAL_PATH / processing_level / subject_name
last_session_path = get_last_session_path(subject_path, subject_name).absolute()
```